src/payment/strategies/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -0,0 +1,144 @@
+import { isHostedInstrumentLike } from '../..';
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType, RequestError } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { StoreCreditActionCreator } from '../../../store-credit';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { StripeElement, StripeElements, StripeUPEClient } from './stripe-upe';
+import StripeUPEScriptLoader from './stripe-upe-script-loader';
+
+export default class StripeUPEPaymentStrategy implements PaymentStrategy {
+    private _stripeUPEClient?: StripeUPEClient;
+    private _stripeElements?: StripeElements;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _stripeScriptLoader: StripeUPEScriptLoader,
+        private _storeCreditActionCreator: StoreCreditActionCreator,
+        private _locale: string
+    ) {}
+
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { stripeupe, methodId, gatewayId } = options;
+        if (!stripeupe?.containerId) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        if (!gatewayId) {
+            throw new InvalidArgumentError('Unable to initialize payment because "gatewayId" argument is not provided.');
+        }
+
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(`${gatewayId}?method=${methodId}`));
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        const { initializationData: { stripePublishableKey, stripeConnectedAccount } } = paymentMethod;
```

Comment:
Not part of you PR, more of a comment for @davidchin or @animesh1987, `initializationData` is typed as `any`, we might want to move that to `unknown` to make sure consumers check what they are receiving... 🤷